### PR TITLE
`Scp330` sound fix

### PR DESCRIPTION
**Pull request by IRacle1**
## Description
**Describe the changes** 
I barely understand what exactly second part of 330interaction transpiler does, but still rewrote it as I found it simpler

**What is the current behavior?** (You can also link to an open issue here)
Surely sever logic is fine, but affects the sound rpc

**What is the new behavior?** (if this is a feature change)
Make ev.Sever work without breaking rpc

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

**Other information**:
@louis1706 @Undid-Iridium im not sure i did that correctly, thats what my patch do:
image

<br />

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [x] I have checked no IL patching errors in the console
image

### Other
- [ ] Still requires more testing

### CODE REVIEW

**Comment by louis1706** [comment]
If you have time could you add a property to chooses whetever or not than sound is made ?

**Review by louis1706 — APPROVED** [review]

File: EXILED/Exiled.Events/Patches/Events/Scp330/InteractingScp330.cs [hunk 1]
```diff
@@ -74,35 +74,60 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
 
-            // This is to find the location of RpcMakeSound to remove the original code and add a new sever logic structure (Start point)
-            int addShouldSeverOffset = -1;
-            int addShouldSeverIndex = newInstructions.FindLastIndex(
-                instruction => instruction.Calls(Method(typeof(Scp330Interobject), nameof(Scp330Interobject.RpcMakeSound)))) + addShouldSeverOffset;
+            /* next code will used to override sound rpc check by EXILED
+             * old:
+             *   if (args.PlaySound)
+             * new:
+             *   if (args.PlaySound || ev.PlaySound)
```
Comment (Misaka-ZeroTwo) [inline EXILED/Exiled.Events/Patches/Events/Scp330/InteractingScp330.cs:81]: `if (args.PlaySound || ev.PlaySound)`

So if you reject it in Exiled, this check is still going to allow play sound?

Comment (IRacle1) [reply EXILED/Exiled.Events/Patches/Events/Scp330/InteractingScp330.cs:81]: Oh, yes, i was trying to make that more nwapi supportive, but ill change that to `&&`

File: EXILED/Exiled.Events/Patches/Events/Scp330/InteractingScp330.cs [hunk 2]
```diff
@@ -74,35 +74,60 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
-            newInstructions[enableEffect].WithLabels(enableEffectLabel);
             newInstructions.InsertRange(
-                addShouldSeverIndex,
+                index,
+                new[]
+                {
+                    // load ev.ShouldPlaySound and or operation with nw property.
+                    new CodeInstruction(OpCodes.Ldloc_S, ev.LocalIndex),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(InteractingScp330EventArgs), nameof(InteractingScp330EventArgs.ShouldPlaySound))),
+                    new(OpCodes.Add),
```
Comment (IRacle1) [inline EXILED/Exiled.Events/Patches/Events/Scp330/InteractingScp330.cs:95]:
```suggestion
                    new(OpCodes.And),
```

**Review by Misaka-ZeroTwo — APPROVED** [review]

**Pull request merged**